AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1326 @@
 #include "azure/core/az_span.h"
 #include "azure/core/az_json.h"
 #include "azure/iot/az_iot_pnp_client.h"
+#include "main.h"
 
-#define DEFAULT_START_TEMP_CELSIUS 22
+#define RETURN_ERR_IF_FAILED(ret)                   \
+  do {                                              \
+    if (az_result_failed(ret)) {                    \
+      return ret;                                   \
+    }                                               \
+  } while (0)
+
+#define RETURN_IF_FAILED(ret)                       \
+  do {                                              \
+    if (az_result_failed(ret)) {                    \
+      return;                                       \
+    }                                               \
+  } while (0)
 
-#define RETURN_ERR_WITH_MESSAGE_IF_FAILED(ret, msg)   \
-  do {                                                \
-    if (az_result_failed(ret)) {                       \
-        debug_printError(msg);                        \
-        return rc;                                    \
-    }                                                 \
+#define RETURN_ERR_WITH_MESSAGE_IF_FAILED(ret, msg) \
+  do {                                              \
+    if (az_result_failed(ret)) {                    \
+      debug_printError(msg);                        \
+      return ret;                                   \
+    }                                               \
   } while (0)
 
 #define RETURN_WITH_MESSAGE_IF_FAILED(ret, msg)     \
   do {                                              \
-    if (az_result_failed(ret)) {                     \
-        debug_printError(msg);                      \
-        return;                                     \
+    if (az_result_failed(ret)) {                    \
+      debug_printError(msg);                        \
+      return;                                       \
     }                                               \
   } while (0)
 
+static int send_reported_property(twin_properties_t* twin_properties);
+
 extern az_iot_pnp_client pnp_client;
 extern pf_MQTT_CLIENT pf_mqtt_iotprovisioning_client;
 extern pf_MQTT_CLIENT pf_mqtt_pnp_client;
 
-// * Plug and Play Values *
-// The model id is the JSON document (also called the Digital Twins Model Identifier or DTMI)
-// which defines the capability of your device. The functionality of the device should match what
-// is described in the corresponding DTMI. Should you choose to program your own PnP capable device,
-// the functionality would need to match the DTMI and you would need to update the below 'model_id'.
-// Please see the sample README for more information on this DTMI.
-const az_span device_model_id = AZ_SPAN_LITERAL_FROM_STR("dtmi:com:example:Thermostat;1");
+// used by led.c to communicate LED states changes
+extern led_status_t led_status;
+
+// used by pin_manager.c
+extern button_press_data_t button_press_data;
+
+extern uint32_t telemetry_interval;
+
+static bool isGetReceived = false;
```

Comment:
please use snake case to be consistent with other var names.
```suggestion
static bool is_get_received = false;
```